### PR TITLE
Remove improper src_mask from encoder tutorial

### LINKED ISSUE #1877: Transformer Language Model Tutorial - Incorrect Attention Mask Description
The tutorial [Language Modeling With nn.Transformer and Torchtext](https://pytorch.org/tutorials/beginner/transformer_tutorial.html) describes an attention mask that will prevent `nn.TransformerEncoder` from attending to not-yet-seen tokens:

> Along with the input sequence, a square attention mask is required because the self-attention layers in nn.TransformerEncoder are only allowed to attend the earlier positions in the sequence.

I think the mask is actually upper triangular: a square mask would not prevent a token attending to future token. 

I may have misunderstood the mask description; if not, happy to write the PR that fixes this.

cc @pytorch/team-text-core @Nayef211

### PR DESCRIPTION
**Pull request by bjhargrave**
Fixes #1877 
The tutorial is using a transformer encoder and the mask used was for masking a decoder which is not part of the tutorial.

The mask is removed. Some variable names are changed to better reflect the purpose of the variable. Also, some unused imports are removed.

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

cc @pytorch/team-text-core @Nayef211

### CODE REVIEW

**Review by Nayef211 — APPROVED** [review]
LGTM. Thanks for helping improve PyTorch tutorials @bjhargrave
